### PR TITLE
Upgrade to Angular 2.0.0rc6, forms 2.0.0rc6 and router 3.0.0rc2

### PR DESCRIPTION
**Pull request by clbond**

### CODE REVIEW

**Comment by clbond** [comment]
@SethDavenport Would love if you could review this for me. Thank you!

**Comment by clbond** [comment]
This work connects to #594

File: webpack/loaders.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
 'use strict';
 
+exports.angular = { // ships in ES6 format now
+  test: /\.js$/,
+  loader: 'babel-loader',
+  include: /angular/,
```
Comment (SethDavenport) [inline webpack/loaders.js]: I'm curious - are they no longer shipping typescript?

File: package.json [hunk 2]
```diff
@@ -45,18 +45,20 @@
...
-    "@angular/router": "^3.0.0-rc.1",
+    "@angular/common": "2.0.0-rc.6",
+    "@angular/compiler": "2.0.0-rc.6",
+    "@angular/core": "2.0.0-rc.6",
+    "@angular/forms": "2.0.0-rc.6",
+    "@angular/http": "2.0.0-rc.6",
+    "@angular/platform-browser": "2.0.0-rc.6",
+    "@angular/platform-browser-dynamic": "2.0.0-rc.6",
+    "@angular/router": "^3.0.0-rc.2",
+    "@angular/testing": "0.0.0-0",
```
Comment (SethDavenport) [inline package.json]: this version number seems odd...

Comment (clbond) [inline package.json]: 0.0.0-0 is the only available version of @angular/testing. (npm info)

File: src/components/alert/alert.component.test.ts [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
-        RioUiModule
-      ]
+  beforeEach(done => {
+    const configure = (testBed: TestBed) => {
+      testBed.configureTestingModule({
+        imports: [RioUiModule],
+      });
+    };
+
+    configureTests(configure).then(testBed => {
```
Comment (SethDavenport) [inline src/components/alert/alert.component.test.ts]: If you return the promise to Jasmine then you don't need the `done` callback

File: webpack/loaders.js (hunk 1 above)
Comment (clbond) [inline webpack/loaders.js]: As far as I know we never used the TypeScript code in our Angular dependencies. We always depended on the .js files. It was just that until now, those files were ES5 and didn't need to be Babel'd in order for us to consume them. If you open up node_modules/@angular and do `find . -name '*.ts'|grep -v '\.d\.ts'`, nothing comes back.

File: src/components/alert/alert.component.test.ts (hunk 3 above)
Comment (clbond) [inline src/components/alert/alert.component.test.ts]: Did not know that. Will update now

Comment (clbond) [inline src/components/alert/alert.component.test.ts]: Fixed

File: webpack/loaders.js (hunk 1 above)
Comment (clbond) [inline webpack/loaders.js]: Sorry, I should clarify. We use the JavaScript output in combination with the .d.ts type definition files. My understanding is that this has always been the case.

File: src/components/alert/alert.component.test.ts (hunk 3 above)
Comment (clbond) [inline src/components/alert/alert.component.test.ts]: Had to revert this ... it broke the tests

Comment (SethDavenport) [inline src/components/alert/alert.component.test.ts]: did you remove done from the artist too?

Comment (clbond) [inline src/components/alert/alert.component.test.ts]: From the artist?

Comment (SethDavenport) [inline src/components/alert/alert.component.test.ts]: arg list.... darn auto incorrect :)

Comment (clbond) [inline src/components/alert/alert.component.test.ts]: Yessir, this was the patch:
[0001-Remove-usage-of-done-and-replace-with-promises-per-n.txt](https://github.com/rangle/angular2-redux-example/files/455850/0001-Remove-usage-of-done-and-replace-with-promises-per-n.txt)

**Comment by clbond** [comment]
Is there anything else wrong with this PR? Can it be merged?

**Pull request merged**